ads_async/asyncio/client.py

```diff
@@ -684,6 +688,146 @@ async def prune_unknown_notifications(self):
             _, port = source
             await self.send(self.circuit.remove_notification(handle), port=port)
 
+    async def download_current_config_file(
+        self,
+        buffer_size: int = 16384,
+    ) -> utils.InMemoryZipFile:
+        """Download the current configuration (CurrentConfig.tszip)."""
+        current_config = await self.get_file(
+            "boot/CurrentConfig.tszip", buffer_size=buffer_size
+        )
+        return utils.InMemoryZipFile(current_config)
+
+    async def download_projects(
+        self,
+        buffer_size: int = 16384,
+    ) -> Dict[str, utils.InMemoryZipFile]:
+        """Download the active project."""
+        current_config = await self.download_current_config_file(
+            buffer_size=buffer_size
+        )
+        projects = {}
+        for fn in current_config.zip.filelist:
+            if fn.filename.startswith("_Config/PLC/"):
+                project_name, _ = os.path.splitext(os.path.basename(fn.filename))
+                tpzip_filename = f"{project_name}.tpzip"
+                projects[tpzip_filename] = utils.InMemoryZipFile(
+                    await self.get_file(
+                        f"boot/CurrentConfig\\{tpzip_filename}", buffer_size=buffer_size
+                    )
+                )
+        return projects
+
+    async def get_file(
+        self,
+        filename: str,
+        encoding: str = constants.ADS_ASYNC_STRING_ENCODING,
+        buffer_size: int = 16384,
+    ) -> bytes:
+        """
+        Download a file from the PLC.
+
+        Parameters
+        ----------
+        filename : str
+            The filename, the start which should be either "boot/" or "target/".
+        directory : DownloadPath
+            The path to download from (/Hard Drive/TwinCAT/3.1/...)
+
+        Returns
+        -------
+        data : bytes
+            The contents of the file.
+
+        Raises
+        ------
+        RequestFailedError
+            With DEVICE_NOTFOUND if the file is not found
+        AdsAsyncException
+            If the download doesn't proceed as expected
+        """
+        try:
+            directory, filename = filename.split("/", 1)
+            path = constants.DownloadPath[directory.lower()]
+        except Exception:
+            raise ValueError(
+                f"The filename should start with boot/ or target/. " f"Got: {filename}"
```

Comment:
Is this a requirement for the ADS interface? Or is this just where are the useful files are?

Reply:
It's because I've only discovered the two possibilities so far based on the traffic I've seen (as it's undocumented).
As-is, these are the only two spots we can download from for now.